components/cdc/src/initializer.rs

```diff
@@ -110,9 +109,11 @@ impl<E: KvEngine> Initializer<E> {
         &mut self,
         change_observer: ChangeObserver,
         cdc_handle: T,
+        concurrency_semaphore: Arc<Semaphore>,
         memory_quota: Arc<MemoryQuota>,
     ) -> Result<()> {
         fail_point!("cdc_before_initialize");
+        let _permit = concurrency_semaphore.acquire().await;
```

Comment:
Do we need to check the `self.downstream_state` after the semaphore is acquired.

Reply:
It's unnecessary becasue we can always get the notification from the Observer.